tool/tbot/main.go

```diff
@@ -227,9 +228,45 @@ func onStart(botConfig *config.BotConfig) error {
 	reloadChan := make(chan struct{})
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-
 	go handleSignals(log, reloadChan, cancel)
 
+	telemetrySentCh := make(chan struct{})
+	go func() {
+		defer close(telemetrySentCh)
+
+		if err := sendTelemetry(
+			ctx, telemetryClient(os.Getenv), os.Getenv, log, botConfig,
+		); err != nil {
+			log.WithError(err).Error(
+				"Failed to send anonymous telemetry.",
+			)
+		}
+	}()
+	// Ensures telemetry finishes sending before function exits.
+	defer func() {
+		select {
+		case <-telemetrySentCh:
+			return
+		case <-ctx.Done():
+		default:
+		}
+
+		waitTime := 30 * time.Second
```

Comment:
30s seems like a long timeout for this call.  I might suggest 5 or 10s in order to fail fast.  If it hasn't succeeded after 10s, I'd reckon there is a 95%+ chance it won't succeed at all.  And I'd rather have a better end user UX in that case than try to capture the long tail of metrics.

Reply:
bec8c1d8a0